main/src/ca/uwaterloo/flix/language/phase/Safety.scala

```diff
@@ -78,6 +78,30 @@ object Safety {
     visitExp(def0.impl.exp, renv, root)
   }
 
+  /**
+    * Checks that if `def0` is a test entry point that it is well-behaved.
+    */
+  private def visitTestEntryPoint(def0: Def, root: Root): List[CompilationMessage] = {
+    def isTest(d: Def): Boolean = d.spec.ann.isTest
+
+    def isUnitType(fparam: FormalParam): Boolean = fparam.tpe.typeConstructor.contains(TypeConstructor.Unit)
+
+    def hasParameters(d: Def, loc: SourceLocation): Boolean = d.spec.fparams match {
+      case fparam :: Nil if isUnitType(fparam) => false
+      case Nil => throw InternalCompilerException("Unexpected empty formal parameter list near", loc)
+      case _ => true
+    }
+
+    root.defs.get(def0.sym).map {
```

Comment:
Why is this lookup needed? Don't we already have the def in hand?

Reply:
Can we reach the spec from there?

Reply:
Ok yeah I see :facepalm: